sysbuild/Kconfig.mcuboot

```diff
@@ -79,27 +79,53 @@ config MCUBOOT_QSPI_XIP_IMAGE_NUMBER
 	default 1 if QSPI_XIP_SPLIT_IMAGE
 	default -1
 
+config MCUBOOT_MCUBOOT_IMAGE_NUMBER
+	int
+	default 4 if SECURE_BOOT_APPCORE && MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 && MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1 && MCUBOOT_QSPI_XIP_IMAGE_NUMBER != 1
+	default 3 if SECURE_BOOT_APPCORE && ((MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1 && (MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 || MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1)) || (MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 && MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1))
+	default 2 if SECURE_BOOT_APPCORE && (MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 || MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1 || MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1)
+	default 1 if SECURE_BOOT_APPCORE
+	default -1
+
 config MCUBOOT_MIN_UPDATEABLE_IMAGES
 	int
 	default 4 if MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 && MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1 && MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1
 	default 3 if (MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 && MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1) || (MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 && MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1) || (MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1 && MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1)
 	default 2 if MCUBOOT_NETWORK_CORE_IMAGE_NUMBER != -1 || MCUBOOT_WIFI_PATCHES_IMAGE_NUMBER != -1 || MCUBOOT_QSPI_XIP_IMAGE_NUMBER != -1
 	default 1
 
+config MCUBOOT_MIN_ADDITIONAL_UPDATEABLE_IMAGES
+	int
+	default 1 if MCUBOOT_MCUBOOT_IMAGE_NUMBER != -1
+	default 0
+
 config MCUBOOT_UPDATEABLE_IMAGES
 	int "Updateable images"
 	range MCUBOOT_MIN_UPDATEABLE_IMAGES MCUBOOT_MAX_UPDATEABLE_IMAGES
-	default 2 if SOC_SERIES_NRF91X && SECURE_BOOT_APPCORE && (MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH || MCUBOOT_MODE_SWAP_SCRATCH || MCUBOOT_MODE_OVERWRITE_ONLY)
+#	default 2 if SOC_SERIES_NRF91X && SECURE_BOOT_APPCORE && (MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH || MCUBOOT_MODE_SWAP_SCRATCH || MCUBOOT_MODE_OVERWRITE_ONLY)
 	help
-	  The number of images that MCUboot will be built with.
+	  The number of images that MCUboot will be built with. Note that if
+	  ``MCUBOOT_ADDITIONAL_UPDATEABLE_IMAGES`` is set to a non-zero value then that value will
+	  be added to the MCUboot updateable image number but not the application, this is to allow
+	  for features like NSIB updates of MCUboot itself.
+
+config MCUBOOT_ADDITIONAL_UPDATEABLE_IMAGES
+	int "Additional MCUboot-only updateable images"
+	range MCUBOOT_MIN_ADDITIONAL_UPDATEABLE_IMAGES MCUBOOT_MIN_ADDITIONAL_UPDATEABLE_IMAGES
+	default MCUBOOT_MIN_ADDITIONAL_UPDATEABLE_IMAGES
+	help
+	  The number of additional images that MCUboot will be built with.
 
 config MCUBOOT_APP_SYNC_UPDATEABLE_IMAGES
 	bool "Sync updateable image value to main application"
-	default n if SOC_SERIES_NRF91X && SECURE_BOOT_APPCORE && (MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH || MCUBOOT_MODE_SWAP_SCRATCH || MCUBOOT_MODE_OVERWRITE_ONLY)
+#	default n if SOC_SERIES_NRF91X && SECURE_BOOT_APPCORE && (MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH || MCUBOOT_MODE_SWAP_SCRATCH || MCUBOOT_MODE_OVERWRITE_ONLY)
```

Comment:
Again why commented-out lines without a comment?

Reply:
Line will be removed in next push

Reply:
Fixed